### PR TITLE
MOB-1288 - Implemented parking list UI 

### PR DESCRIPTION
**Pull request by Oleg-Pecheneg**

### CODE REVIEW

File: unstoppable-ios-app/domains-manager-ios/Extensions/UIImage.swift [hunk 1]
```diff
@@ -121,6 +121,7 @@ extension UIImage {
     static let arrowRight = UIImage(named: "arrowRight")!
     static let timeIcon24 = UIImage(named: "timeIcon24")!
     static let dragIcon24 = UIImage(named: "dragIcon24")!
+    static let parkingIcon24 = UIImage(named: "parkingIcon24")!
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Extensions/UIImage.swift:124]: Maybe extract these icons?

File: unstoppable-ios-app/domains-manager-ios/Modules/LoginViewController/LoginViewPresenter.swift [hunk 2]
```diff
@@ -92,8 +93,17 @@ private extension LoginViewPresenter {
     }
     
     @MainActor
-    func userAuthorized() {
-        view?.cNavigationController?.popViewController(animated: true)
+    func userAuthorized() async {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/LoginViewController/LoginViewPresenter.swift:96]: It means `didUserAuthorize` or `isUserAuthorized`?

File: unstoppable-ios-app/domains-manager-ios/Modules/Settings/SettingsPresenter.swift [hunk 3]
```diff
@@ -226,14 +229,23 @@ private extension SettingsPresenter {
...
             Task {
                 do {
                     try await appContext.pullUpViewService.showLogoutConfirmationPullUp(in: view)
                     await view.dismissPullUpMenu()
                     try await appContext.authentificationService.verifyWith(uiHandler: view, purpose: .confirm)
-                    FirebaseInteractionService.shared.logout()
+                    firebaseInteractionService.logout()
                 } catch { }
+                                
+//                do {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Modules/Settings/SettingsPresenter.swift:241]: Dead?

File: unstoppable-ios-app/domains-manager-ios/Services/Authorisation/FirebaseInteractionService.swift [hunk 4]
```diff
@@ -86,20 +101,23 @@ extension FirebaseInteractionService {
...
+        var result = [FirebaseDomain]()
+        var isThereDomainsToLoad = true
+        var page = 1
+        let perPage = 200
         
-        let url = URL(string: "\(baseAPIURL())user/domains?extension=All&page=1&perPage=50&status=all")!
-        let request = APIRequest(url: url, body: "", method: .get)
-        let response: Response = try await makeFirebaseDecodableAPIDataRequest(request,
-                                                                               dateDecodingStrategy: .defaultDateDecodingStrategy())
+        while isThereDomainsToLoad {
```
Comment (rommex) [inline unstoppable-ios-app/domains-manager-ios/Services/Authorisation/FirebaseInteractionService.swift:109]: Would be nice to use `protocol PaginatedFetchable` ?

**Review by rommex — APPROVED** [review]
With questions

File: unstoppable-ios-app/domains-manager-ios/Modules/LoginViewController/LoginViewPresenter.swift (hunk 2 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Modules/LoginViewController/LoginViewPresenter.swift:96]: Updated to userDidAuthorize()

File: unstoppable-ios-app/domains-manager-ios/Extensions/UIImage.swift (hunk 1 above)
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Extensions/UIImage.swift:124]: This is place where all extracted icons placed

File: unstoppable-ios-app/domains-manager-ios/Entities/Domains/FirebaseDomainDisplayInfo.swift [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import Foundation
+
+struct FirebaseDomainDisplayInfo: Codable, Hashable {
+    
+    var claimStatus: String
+    var internalCustody: Bool
+    var purchasedAt: Date?
+    var parkingExpiresAt: Date?
+    var blockchain: String
```
[the remark this answers is not in the document]
Comment (Oleg-Pecheneg) [reply unstoppable-ios-app/domains-manager-ios/Entities/Domains/FirebaseDomainDisplayInfo.swift:16]: I would like to make it too, but i don't feel at this step our API is solid enough to use enum

**Pull request merged**